docs/prune.md

```diff
@@ -0,0 +1,186 @@
+# prune directive in kustomization.yaml
+
+Since Kustomize 2.1.0, a new field `prune` is added and can be used as the following in kustomization.yaml:
+```yaml
+prune:
+  type: ConfigMap
+  configMap:
+    name: prune-cm-name
+    namespace: some-namespace
+```
+
+### Motivation
```

Comment:
This section reads as more of an implementation discussion.  How about

> ### Motivation
> This ConfigMap object can be consumed by a client such as those under development in
> [cli-experimental](https://github.com/kubernetes-sigs/cli-experimental).
> The client can recognize this object by name and use it to do a better job with
> actions like `apply`, `prune` and `delete`.
>
> Read more about this [below](#how-prune-works)
>
> ### Implementation

stuff below